File: src/Homeworks/Homework1/Parser.cs
using System;
using System.Linq;

namespace Homework1
{
	public static class Parser
	{
		private static readonly string[] SupportedOperations = {"+", "-", "*", "/"};

		public static int TryParseArguments(string[] args, out int val1, out string operation, out int val2)
		{
			var isVal1Int = int.TryParse(args[0], out val1);
			operation = args[1];
			var isVal2Int = int.TryParse(args[2], out val2);

			if (!isVal1Int || !isVal2Int)
			{
				Console.WriteLine($"{args[0]}{args[1]}{args[2]} is not a valid calculate syntax");
				return 1;
			}

			if (!SupportedOperations.Contains(operation))
			{
				Console.WriteLine($"{args[0]}{args[1]}{args[2]} is not a valid calculate syntax" +
				                  $"\nSupported operations are:{SupportedOperations.Aggregate((x, y) => x + " " + y)}");
				return 2;
			}

			return 0;
		}
	}
}

File: src/Homeworks/Homework4/Parser.fs
module Homework4.Parser

open System

let SupportedOperations = [|"+"; "-"; "*"; "/"; ":"|]

let TryParseArguments (args: string[]) (val1: byref<int>) (operation: outref<string>) (val2: outref<int>) =
    let isVal1Int = Int32.TryParse(args.[0], &val1)
    let isVal2Int = Int32.TryParse(args.[2], &val2)
    operation <- args.[1]
    if (isVal1Int && isVal2Int) = false then
        printf $"{args.[0]}{args.[1]}{args.[2]} is not a valid calculate syntax"
        1
    else if Array.contains operation SupportedOperations = false then
        printf $"{args.[0]}{args.[1]}{args.[2]} is not a valid calculate syntax \n Supported operations are + - * / :"
        2
    else 0
   
 
    